setup.py

```diff
@@ -73,13 +73,14 @@
     "mypy>=0.812",
     "pydocstyle>=6.1.1",
     # Docs
-    "sphinx<3.5.0",
-    "sphinx-rtd-theme==0.4.3",
+    "sphinx>=4.0.0",
     "sphinxemoji>=0.1.8",
     "sphinx-copybutton>=0.3.1",
     "docutils<0.18",
     "recommonmark>=0.7.1",
     "sphinx-markdown-tables>=0.0.15",
+    "sphinx-tabs>=3.3.0",
```

Comment:
sphinx tabs can be removed in this PR btw (let's add it in the relevant PRs that use tabs for docstrings)